pkg/mover/chart.go

```diff
@@ -230,15 +224,65 @@ func (cm *ChartMover) Print() {
 	cm.printMove()
 }
 
-func (cm *ChartMover) loadChart(chartPath string) error {
+func (cm *ChartMover) loadChart(src *Source) error {
 	var err error
+	if src.Chart.Local != nil {
+		cm.chart, err = loader.Load(src.Chart.Local.Path)
+		if err != nil {
+			return &ChartLoadingError{Path: src.Chart.Local.Path, Inner: err}
+		}
+		return nil
+	} else if src.Chart.IntermediateBundle != nil {
+		return cm.loadChartFromIntermediateBundle(src.Chart.IntermediateBundle.Path)
+	}
+	return fmt.Errorf("must provide either a local chart or an intermediate bundle as input")
+}
+
+func (cm *ChartMover) loadChartFromIntermediateBundle(bundlePath string) error {
+	var err error
+	cm.bundle, err = openBundle(bundlePath)
+	if err != nil {
+		return err
+	}
+
+	chartPath, err := os.MkdirTemp("", "bundle-chart-*")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory to extract bundled chart %s: %w",
+			cm.bundle.bundlePath, err)
+	}
+	defer os.RemoveAll(chartPath)
+
+	if err := cm.bundle.ExtractChartTo(chartPath); err != nil {
+		return err
+	}
+
 	cm.chart, err = loader.Load(chartPath)
 	if err != nil {
 		return &ChartLoadingError{Path: chartPath, Inner: err}
 	}
 	return nil
 }
 
+func (cm *ChartMover) loadHints(src *Source) error {
+	var err error
+	if src.Chart.IntermediateBundle != nil {
+		if src.ImageHintsFile != "" {
+			return fmt.Errorf("intermediate bundles already embed the hints file" +
+				", skip explicit hints, they will be ignored")
+		}
+		cm.rawHints, err = cm.bundle.LoadHints(cm.logger)
+	} else {
+		cm.rawHints, err = loadHints(src.ImageHintsFile, cm.chart, cm.logger)
+	}
+	if err != nil {
+		return err
+	}
+	if cm.rawHints == nil {
+		return ErrImageHintsMissing
```

Comment:
Would this mean that the hints is missing or that's empty? If the latter we might need to adapt/change the error type/message.

Reply:
At this point it should be missing.
If it is empty we should have complained before in the load operation instead.

Reply:
ensuring we fail differently if empty at load time...

Reply:
done